appendable-rs/appendable/src/handler/jsonl_handler.rs

```diff
@@ -0,0 +1,71 @@
+use crate::index_file::Index;
+use crate::io::DataHandler;
+use serde_json::{Deserializer, Map, Value};
+use std::fs::File;
+use std::io::{BufRead, BufReader, Seek, SeekFrom};
+use xxhash_rust::xxh3::Xxh3;
+
+pub struct JSONLHandler {
+    reader: BufReader<File>,
+    xxh3: Xxh3,
+}
+impl JSONLHandler {
+    pub fn new(file: File) -> Self {
```

Comment:
Can we use a `Cursor` instead of `File` here to also allow non-filesystem-backed implementations? https://doc.rust-lang.org/std/io/struct.Cursor.html

Reply:
Oh awesome -- most definitely!